library/spdm_requester_lib/libspdm_req_set_key_pair_info.c

```diff
@@ -0,0 +1,219 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "internal/libspdm_requester_lib.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_SET_KEY_PAIR_INFO_CAP
+
+/**
+ * This function sends SET_KEY_PAIR_INFO and receives SET_KEY_PAIR_INFO_ACK
+ *
+ * @param  context             A pointer to the SPDM context.
+ *
+ **/
+static libspdm_return_t libspdm_try_set_key_pair_info(libspdm_context_t *spdm_context,
+                                                      const uint32_t *session_id,
+                                                      uint8_t key_pair_id,
+                                                      uint8_t operation,
+                                                      uint16_t desired_key_usage,
+                                                      uint32_t desired_asym_algo,
+                                                      uint8_t desired_assoc_cert_slot_mask
+                                                      )
+{
+    libspdm_return_t status;
+    spdm_set_key_pair_info_request_t *spdm_request;
+    size_t spdm_request_size;
+    spdm_set_key_pair_info_ack_response_t *spdm_response;
+    size_t spdm_response_size;
+    uint8_t *message;
+    size_t message_size;
+    size_t transport_header_size;
+    libspdm_session_info_t *session_info;
+    libspdm_session_state_t session_state;
+    uint8_t *ptr;
+
+    /* -=[Check Parameters Phase]=- */
+    if (libspdm_get_connection_version(spdm_context) < SPDM_MESSAGE_VERSION_13) {
+        return LIBSPDM_STATUS_UNSUPPORTED_CAP;
+    }
+
+    if (key_pair_id == 0) {
+        return LIBSPDM_STATUS_INVALID_PARAMETER;
+    }
+
+    if (operation > SPDM_SET_KEY_PAIR_INFO_GENERATE_OPERATION) {
+        return LIBSPDM_STATUS_INVALID_PARAMETER;
+    }
+    if (operation == SPDM_SET_KEY_PAIR_INFO_ERASE_OPERATION) {
+        if ((desired_key_usage != 0) || (desired_asym_algo != 0) ||
+            (desired_assoc_cert_slot_mask != 0)) {
+            return LIBSPDM_STATUS_INVALID_PARAMETER;
+        }
+    }
+    if (operation == SPDM_SET_KEY_PAIR_INFO_GENERATE_OPERATION) {
```

Comment:
This check is not needed. The Requester can ask the Responder to generate a key without having `DesiredKeyUsage`, `DesiredAsymAlgo`, or `DesiredAssocCertSlotMask` set. The Requester can later use the `ParameterChange` operation to set these.

Reply:
agree.